docs_guidelines/tutorials/new_doc_project.rst

```diff
@@ -0,0 +1,610 @@
+This matrix will be called ``python-version`` and its value will be ``[3.8]``. 
+
+
+.. code-block:: yaml
+
+    jobs:
+      docs_publish:
+        if: ${{ startsWith(github.ref, 'refs/heads/stable') && contains('["mantainer-1","mantainer-2", ...]', github.actor) }}
+        runs-on: ubuntu-latest
+        strategy:
+          matrix:
+            python-version: [3.8]
+        steps:
+
+Steps
+^^^^^
+
+Check-out the repository
+"""""""""""""""""""""""""
+
+The first step of this job consists of enabling the workflow to access the repository ``qiskit-x``. To do that you only need to call the latest version (version 3, ``@v3``) the pre-defined `checkout action <https://github.com/actions/checkout>`_ with the `uses <https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsuses>`_ key.
+By setting the ``fetch-depth`` parameter to ``0`` via the `with <https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idwith>`_ key, you can let the action reach for the full history of your repository.
+
+.. code-block:: yaml
+
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+
+Setup Python
+""""""""""""
+
+Once your repo has been reached, the workflow continues by installing a Python version, in this case, the 3.8, as defined in ``matrix.python-version``. Fortunately, the
+`setup-python action <https://github.com/actions/setup-python>`_ already does this for us!. To choose the Python version you only have to set the ``python-version`` parameter with the corresponding value.
+
+.. code-block:: yaml
+
+    - uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+
+
+Install the package
+""""""""""""""""""""
+
+Now that you have Python, let's install our package, Qiskit X and its more immediate dependencies and constraints from ``requirements-dev.txt`` and ``constraints.txt`` respectively. 
+You can use the `run <https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun>`_ key and add the usual bash commands as a multiline block with the pipe indicator ``|``. These commands are: ``pip install -e .`` for the editable (``-e``) version of the package, ``pip install -U -r requirements-dev.txt`` for the latest version (``-U``) of the requirements (``-r``) and ``pip install -c constraints.txt`` for the constraints (``-c``).
+You can set the shell to ``bash`` via the
+`shell <https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell>`_ keyword.
+
+
+.. code-block:: yaml
+
+    - name: Install X
+      run: |
+        pip install -e .
+        pip install -U -c constraints.txt -r requirements-dev.txt
+      shell: bash
+
+
+Install other dependencies
+""""""""""""""""""""""""""
+
+In case there are another dependencies that are not covered in the ``requirements-dev.txt``, you can add them in a similar step.
```

Comment:
Can you add some examples for these kind of dependencies that are not in `requirements-dev.txt`? I think it can be non-python dependencies like`pandoc`, `graphviz` but I think a python dependency is even more interesting. Perhaps it dependencies that are only used for docs building not the development. But often those dependencies are already included in `requirements-dev.txt` as well. For example, https://github.com/Qiskit/qiskit/blob/master/requirements-dev.txt includes sphinx extensions like `sphinx-design`, `sphinx-gallery`.

Reply:
Sure. I can look for some examples from say `qiskit-nature` or `qiskit-machine-learning`.

Reply:
Included some examples of both python and non-python packages from ``qiskit-nature`` and ``qiskit-machine-learning``. In particular ``torchvision`` (python), ``pandoc`` and ``graphviz`` (non-python).